### PR TITLE
#1979 Migrate project to pnpm

### PR DESCRIPTION
**Pull request by SanderKnauff**
This PR addresses issue #1979

pnpm has been chosen over another package manager because the Ember ecosystem seems to be using it over `npm` or newer versions of `yarn`.

For the purposes of keeping development and CI tooling in sync, Volta has also been added to the `package.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "@glimmer/component": "1.1.2",
     "@glimmer/tracking": "1.1.2",
     "@release-it-plugins/lerna-changelog": "6.0.0",
-    "bootstrap": "5.3.2",
+    "bootstrap": "^5.3.2",
```
Comment (jelhan) [inline package.json:83]: Why are you changing the bootstrap dependency to a version range instead of a pinned version?

**Review by jelhan — CHANGES_REQUESTED** [review]
Thanks a lot for picking this up!

It seems you missed deleting the `docs/yarn.lock`. Can you please double check?

And we should configure RenovateBot to dedupe dependencies with pnpm. Deduplicating dependencies for yarn is done by the shared preset: https://github.com/kaliber5/renovate-config/blob/e63aaa1ed189d3aedfcbe79e909e80c7f26bcac2/default.json#L29 For PNPM we need to add the configuration ourselves. RenovateBot documentation on the option is available here: https://docs.renovatebot.com/configuration-options/#postupdateoptions

I assume GitHub actions CI workflow is based on Ember CLI blueprints, correct?

File: package.json (hunk 1 above)
Comment (SanderKnauff) [reply package.json]: Ah, that must have happened when I tried fixing the `package.json` after a local ember-try run went wrong. I've reverted it.

**Comment by SanderKnauff** [comment]
> Thanks a lot for picking this up!
> 
> It seems you missed deleting the `docs/yarn.lock`. Can you please double check?
> 
> And we should configure RenovateBot to dedupe dependencies with pnpm. Deduplicating dependencies for yarn is done by the shared preset: https://github.com/kaliber5/renovate-config/blob/e63aaa1ed189d3aedfcbe79e909e80c7f26bcac2/default.json#L29 For PNPM we need to add the configuration ourselves. RenovateBot documentation on the option is available here: https://docs.renovatebot.com/configuration-options/#postupdateoptions
> 
> I assume GitHub actions CI workflow is based on Ember CLI blueprints, correct?

Partially, I've tried to keep the current config as much as possible as I was not sure of the implications of larger changes.
The biggest change might be that the `@wyvox/action` handles the checkout and workspace setup.

I've also added the dedupe to `renovate.json` but kept the current preset so we don't have to copy the entire file.

**Review by jelhan — APPROVED** [review]
Thanks a lot!

**Pull request merged**